source/Irrlicht/CBawFile.h

```diff
@@ -66,11 +78,26 @@ namespace irr { namespace core
 	};
 
 #include "irrpack.h"
-	//! Utility struct. Used only while loading (CBAWMeshLoader). Cast blob pointer to MeshBlob* to make life easier.
-	struct MeshBlob : Blob
+	//! Utility struct. Cast blob pointer to MeshBlob* to make life easier.
+	struct MeshBlobV1 : Blob
 	{
+		//! WARNING: Constructor saves only bounding box and mesh buffer count (not mesh buffer pointers)
+		explicit MeshBlobV1(const aabbox3df & _box, uint32_t _cnt);
+
+		aabbox3df box;
 		uint32_t meshBufCnt;
+		uint64_t meshBufPtrs[1];
+	} PACK_STRUCT;
+
+	//! Utility struct. Cast blob pointer to MeshBlob* to make life easier.
+	struct SkinnedMeshBlobV1 : Blob
+	{
+		//! WARNING: Constructor saves only bone hierarchy, bounding box and mesh buffer count (not mesh buffer pointers)
+		explicit SkinnedMeshBlobV1(scene::CFinalBoneHierarchy* _fbh, const aabbox3df & _box, uint32_t _cnt);
```

Comment:
Maybe have a constructor/static create function directly from the object being saved

+see my note about variable size blobs

Reply:
>Maybe have a constructor/static create function directly from the object being saved
  
We already talked about this and decided to load blob buffer with variable-size data outside ctor since it has to rely on the fact that the memory actually is allocated (and check it from inside ctor) and it's kinda dangerous